sql-plugin/src/main/340+/scala/com/nvidia/spark/rapids/shims/TypeUtilsShims.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.types.{DataType, NullType, NumericType}
+
+/**
+ * Reimplement the function `checkForNumericExpr` which has been removed since
+ * Spark 3.4.0
+ */
+object TypeUtilsShims {
+  def checkForNumericExpr(dt: DataType, caller: String): TypeCheckResult = {
```

Comment:
Spark is using `checkForAnsiIntervalOrNumericType` now instead of `checkForNumericExpr`.  So `checkForNumericExpr` is removed. I am okay adding this in 340+/ shim until we have ANSI support for aggregate functions - https://github.com/NVIDIA/spark-rapids/issues/5114 . 
@tgravescs - Please let us know if you think otherwise.

Reply:
yes, agree, we can add Ansi support with the other issue.